ddtrace/internal/_encoding.pyx

```diff
@@ -166,22 +166,27 @@ cdef class StringTable(object):
     cdef insert(self, object string):
         pass
 
-    cdef stdint.uint32_t _index(self, object string):
+    cdef stdint.uint32_t _index(self, object string) except? -1:
         cdef stdint.uint32_t _id
+        cdef int ret
 
         if string is None:
             return 0
 
-        if PyDict_Contains(self._table, string):
-            return PyLong_AsLong(<object>PyDict_GetItem(self._table, string))
+        ret = PyDict_Contains(self._table, string)
+        if ret == -1: return ret
+        if ret:
+            ret = PyLong_AsLong(<object>PyDict_GetItem(self._table, string))
```

Comment:
```suggestion
            return PyLong_AsLong(<object>PyDict_GetItem(self._table, string))
```

we can revert this since if `PyLong_AsLong` returns `-1` then we want to return anyway.